### ISSUE TITLE
Limit how often the mempool revalidates its contents during bulk sync

### ISSUE THREAD

**Issue by nfrisby** [input-output-hk/ouroboros-network#3029]
The mempool currently revalidates its contents whenever the current ledger state's tip point changes. During bulk sync, that happens ~hundreds of times per second. So if the mempool were somehow significantly non-empty, that could be costly overhead.

In contrast, once the node has caught up, the ledger tip should instead only evolve once per several seconds (on mainnet, at least).

1) We're not sure it would be worthwhile to rate-limit this overhead, but 2) it seems like it could indeed be important to do so. So it's worth asking the question this Issue represents. EG If the mempool is large, could revalidating it for each new block (~3 times per minutes when caught up) be "too often"?

**Comment by dcoutts** [comment]
This is an excellent point, and should not be too hard to do. We already update the mempool asynchronously, so it should be easy enough to have some logic there to rate limit. And indeed we can know if we're syncing or not and use that to decide rate limiting (or perhaps more extreme: if we forward txs at all, since most will be invalid to us when we're still syncing).

**Comment by nfrisby** [comment]
> So if the mempool were somehow significantly non-empty

Thinking about this today, we don't have any ideas about how we'd reach this state, at least assuming honest behavior.

**Comment by nfrisby** [comment]
We chatted with Duncan. Current plan: take this observation farther and "pause" the TxSub protocol while the node think's it is catching up. EG Use the same logic as in `readFetchMode` implemented in `NodeKernel.hs`.

TODO: repurpose this Issue -- or else Close it and open a new one.

**Comment by edsko** [comment]
Note that a _resttarted_ node (relay or core not) will always have an empty mempool because the mempool is emphemeral.

**Comment by dcoutts** [comment]
Yes, so having discussed this a bit more we think the right thing to do is to enable/disable the inbound side of the tx submission based on whether consensus thinks it is behind.

This would both avoid wasted effort (validating txs that will almost certainly fail because we're behind) and would also avoid us accumulating bogus statistics about how many invalid txs our peers are sending us (which we will use at some stage to add a bit more DoS resistance to the transaction relay scheme).
